tests/test_control_flow.py

```diff
@@ -0,0 +1,208 @@
+from unittest.mock import patch
+
+import torch
+
+import torchdynamo.testing
+from torchdynamo.logic.control_flow import cond
+
+
+class ControlFlowTests(torchdynamo.testing.TestCase):
+    @patch.object(torchdynamo.config, "fake_tensor_propagation", False)
+    def test_simple_condition(self):
+        from_true = None
+        from_false = None
+
+        @torchdynamo.optimize("eager", nopython=True)
+        def foo():
+            nonlocal from_true
+            nonlocal from_false
+
+            def compute(pred, z):
+                def when_true(x):
+                    return x + x
+
+                def when_false(x):
+                    return x * x
```

Comment:
Does it still work if `when_true` or `when_false` capture values that are not directly passed into the function?

Reply:
No - we reject closures in dynamo as unsupported.

Reply:
In general dynamo does support closures, but not for export.

Reply:
Yes. Anything unsupported becomes a graph break, falling back to eager. In export, we do whole capture or nothing. Currently in this PR the true and false fn are captured with export.